wrappers/src/shared_realm_cs.cpp

```diff
@@ -39,13 +39,20 @@ using SharedAsyncOpenTask = std::shared_ptr<AsyncOpenTask>;
 using namespace realm;
 using namespace realm::binding;
 
+using OpenRealmCallbackT = void(void* task_completion_source, ThreadSafeReference* ref, int32_t error_code, const char* message, size_t message_len);
+using RealmChangedT = void(void* managed_state_handle);
+using GetNativeSchemaT = void(SchemaForMarshaling schema, void* managed_callback);
+using OnBindingContextDestructedT = void(void* managed_handle);
+using LogMessageT = void(realm_value_t message, util::Logger::Level level);
+
 namespace realm {
 namespace binding {
-    void (*s_open_realm_callback)(void* task_completion_source, ThreadSafeReference* ref, int32_t error_code, const char* message, size_t message_len);
-    void (*s_realm_changed)(void* managed_state_handle);
-    void (*s_get_native_schema)(SchemaForMarshaling schema, void* managed_callback);
-    void (*s_on_binding_context_destructed)(void* managed_handle);
-    void (*s_log_message)(realm_value_t message, util::Logger::Level level);
+    std::function<void(void* task_completion_source, ThreadSafeReference* ref, int32_t error_code, const char* message, size_t message_len)> s_open_realm_callback;
+    std::function<void(void* managed_state_handle)> s_realm_changed;
+    std::function<void(SchemaForMarshaling schema, void* managed_callback)> s_get_native_schema;
+    std::function<void(void* managed_handle)> s_on_binding_context_destructed;
+    std::function<void(realm_value_t message, util::Logger::Level level)> s_log_message;
+    std::atomic<bool> s_can_call_managed;
```

Comment:
This will be `false` if the C# domain has been unloaded (thus it's not safe to invoke managed callbacks). It's set to `false` in `shared_realm_close_all_realms` which is invoked on domain unload and back to `true` in `shared_realm_install_callbacks`, which is invoked whenever the SharedRealmHandle is initialized.